### ISSUE TITLE
why x_hat reqires_grad?

### ISSUE THREAD

**Issue by kaca0083** [jfsantos/dragan-pytorch#1]
x_hat = Variable(alpha * X.data + (1 - alpha) * (X.data + 0.5 * X.data.std() * torch.rand(X.size())), requires_grad=True)
I suppose we don't need the gradients x_hat right? Since x_hat is only related to X.data.
